### PR TITLE
Check for enums as well as scalars when doing UNWIND CREATE

### LINKED ISSUE #2889: Unable to insert list of with mutations
**Describe the bug**
Hi. I am trying to insert mandatory list of enums via mutations but it files. 

**Type definitions**
Minimal example types:

```graphql
enum MyEnum {
    FIRST
    SECOND
}

type MyEnumHolder {
    myEnums: [MyEnum!]!
}
```

Mutation query:

```graphql
mutation {
  createMyEnumHolders(input: [{
    myEnums: [FIRST, SECOND]
  }]) {
    myEnumHolders {
      myEnums
    }
  } 
}
```

Error I am getting 

`"Cannot return null for non-nullable field MyEnumHolder.myEnums."`

**To Reproduce**
Neo4j where is running Apollo server running  

**Screenshots**
Screenshot 2023-02-15 at 11 20 32

* OS: macOS but server is running in docker `neo4j:4.4.14-community` 

**Additional context**
If I query db after failed insert attempt it contains node without enums list

Screenshot 2023-02-15 at 11 28 28

And if I try to create item via cypher query it is OK.

Screenshot 2023-02-15 at 11 29 26
Screenshot 2023-02-15 at 11 29 35

### PR DESCRIPTION
**Pull request by darrellwarde**
# Description

> **Note**
> 
> Please provide a description of the work completed in this PR below

When generating Cypher for an `UNWIND CREATE`, input should have been checked for enums as well as scalars.

## Complexity

> **Note**
>
> Please provide an estimated complexity of this PR of either Low, Medium or High

Complexity: Low

# Issue

> **Note**
>
>  Please link to the GitHub issue(s) in which the proposal for this work was discussed
>  
>  To link to multiple issues, use full syntax for each, for example `Closes #1, closes #2, closes #3`

Closes #2889

### CODE REVIEW

**Comment by neo4j-team-graphql** [comment]
Made some changes to the GraphQL Toolbox? Or made some changes to the core library perhaps?

Check out your changes at https://neo4j-graphql-toolbox-stage-2913.surge.sh

Use for instance a local dbms hosted on [Neo4j Desktop](https://neo4j.com/download/) or a cloud-hosted [Aura Free](https://neo4j.com/cloud/aura-free/) dbms.

**Comment by neo4j-team-graphql** [comment]
# Performance Report

No Performance Changes


<details>
    <summary>Show Full Table</summary>
    
| name | dbHits | old dbHits | time (ms) | old time (ms) | maxRows |
| ---- | ------ | ---------- | --------- | ------------- | ------- |
|  aggregations.TopLevelAggregate | 3403 | 3403 | 44 | 60 | 1134 |
|  aggregations.NestedAggregation | 13514 | 13514 | 108 | 134 | 2174 |
|  aggregations.AggregationWithWhere | 10942 | 10942 | 64 | 94 | 2174 |
|  aggregations.AggregationWhereWithinNestedRelationships | 20101954 | 20101954 | 3725 | 3586 | 2008534 |
|  aggregations.AggregationWhereWithinNestedConnections | 20101954 | 20101954 | 3203 | 3173 | 2008534 |
|  aggregations.NestedCountFromMovieToActors | 8694 | 8694 | 62 | 126 | 2174 |
|  aggregations.NestedCountFromActorsToMovie | 8900 | 8900 | 97 | 102 | 2174 |
|  aggregations.DeeplyNestedCount | 10054408 | 10054408 | 5630 | 5143 | 2008534 |
|  batch-create.BatchCreate | 3600 | 3600 | 102 | 123 | 600 |
|  connect.createAndConnect | 12419 | 12419 | 299 | 318 | 3003 |
|  connections.Connection | 13042 | 13042 | 238 | 161 | 2174 |
|  connections.NestedConnection | 33883 | 33883 | 156 | 157 | 4516 |
|  connections.ConnectionWithSort | 2277 | 2277 | 90 | 102 | 1040 |
|  connections.ConnectionWithSortAndCypher | 2277 | 2277 | 240 | 388 | 1040 |
|  create.SimpleMutation | 6 | 6 | 43 | 58 | 1 |
|  cypher-directive.TopLevelCypherDirective | 99525 | 99525 | 818 | 851 | 10168 |
|  cypher-directive.TopLevelCypherDirectiveWithColumnName | 127994 | 127994 | 316 | 305 | 10168 |
|  delete.SimpleDelete | 18361 | 18361 | 408 | 279 | 1040 |
|  delete.NestedDeleteInUpdate | 16779 | 16779 | 244 | 165 | 2040 |
|  query.SimpleQuery | 14120 | 14120 | 134 | 135 | 2174 |
|  query.QueryWhere | 8675 | 8675 | 44 | 53 | 2166 |
|  query.SimpleQueryWithNestedWhere | 8846 | 8846 | 58 | 74 | 2166 |
|  query.Nested | 10084290 | 10084290 | 12116 | 12302 | 2008534 |
|  query.NestedWithFilter | 37172 | 37172 | 172 | 165 | 4511 |
|  query.OrFilterOnRelationships | 37454 | 35653 | 201 | 238 | 1971 |
|  query.OrFilterOnRelationshipsAndNested | 30889 | 29314 | 283 | 274 | 1971 |
|  query.QueryWithNestedIn | 12536 | 12695 | 69 | 90 | 1280 |
|  query.NestedConnectionWhere | 8794 | 8794 | 68 | 82 | 2174 |
|  query.DeeplyNestedConnectionWhere | 9004 | 9004 | 106 | 114 | 2174 |
|  query.DeeplyNestedWithRelationshipFilters | 6251 | 6251 | 221 | 178 | 1134 |
|  query.NestedWithRelationshipSingleFilters | 3021189 | 3021189 | 768 | 792 | 1003942 |
|  query.Fulltext | 96 | 96 | 45 | 36 | 16 |
|  query.FulltextWithNestedQuery | 571 | 571 | 52 | 57 | 84 |
|  sorting.SortMultipleTypes | 2493 | 2493 | 82 | 87 | 1040 |
|  sorting.SortMultipleTypesWithCypherWithCypher | 1439 | 1439 | 211 | 281 | 1040 |
|  sorting.SortOnNestedFields | 13042 | 13042 | 100 | 95 | 2174 |
|  sorting.SortDeeplyNestedFields | 38965 | 38965 | 132 | 147 | 4516 |
|  sorting.SortWithTopLevelCypher | 2119 | 2119 | 125 | 200 | 1040 |
|  unions.SimpleUnionQuery | 321 | 321 | 69 | 87 | 35 |
|  unions.SimpleUnionQueryWithMissingFields | 293 | 293 | 75 | 74 | 35 |
|  unions.NestedUnion | 398553 | 398553 | 349 | 640 | 33033 |
|  unions.NestedUnionWithMissingFields | 372527 | 372527 | 463 | 395 | 33033 |
|  update.NestedUpdate | 2119 | 2119 | 55 | 66 | 1040 |

</details>
        
Old Schema Generation: 39.407s
Schema Generation: 38.712s

File: packages/graphql/src/translate/batch-create/parser.ts [hunk 1]
```diff
@@ -106,13 +106,14 @@ export function inputTreeToCypherMap(
     return new Cypher.Map(properties);
 }
 
-function isScalar(fieldName: string, graphElement: GraphElement) {
+function isScalarOrEnum(fieldName: string, graphElement: GraphElement) {
     const scalarPredicate = (x) => x.fieldName === fieldName;
     const scalarFields = [
```
Comment (Liam-Doodson) [inline packages/graphql/src/translate/batch-create/parser.ts:111]:
```suggestion
    const scalarOrEnumFields = [
```

File: packages/graphql/src/translate/batch-create/parser.ts [hunk 2]
```diff
@@ -106,13 +106,14 @@ export function inputTreeToCypherMap(
     return new Cypher.Map(properties);
 }
 
-function isScalar(fieldName: string, graphElement: GraphElement) {
+function isScalarOrEnum(fieldName: string, graphElement: GraphElement) {
     const scalarPredicate = (x) => x.fieldName === fieldName;
```
Comment (Liam-Doodson) [inline packages/graphql/src/translate/batch-create/parser.ts:110]:
```suggestion
    const scalarOrEnumPredicate = (x) => x.fieldName === fieldName;
```

File: packages/graphql/src/translate/batch-create/parser.ts [hunk 3]
```diff
@@ -106,13 +106,14 @@ export function inputTreeToCypherMap(
...
     const scalarFields = [
         graphElement.primitiveFields,
         graphElement.temporalFields,
         graphElement.pointFields,
         graphElement.scalarFields,
+        graphElement.enumFields
     ];
     return scalarFields.flat().some(scalarPredicate);
```
Comment (Liam-Doodson) [inline packages/graphql/src/translate/batch-create/parser.ts:118]:
```suggestion
    return scalarOrEnumFields.flat().some(scalarOrEnumPredicate);
```

**Review by Liam-Doodson — APPROVED** [review]
Looks good other than me being nit-picky on variable names!

**Review by MacondoExpress — APPROVED** [review]
LGTM, thank you for the fix!

**Pull request merged**